lightning/src/onion_message/messenger.rs

```diff
@@ -689,14 +689,20 @@ where
 				});
 
 			match node_details {
-				Some((features, addresses))
-					if features.supports_onion_messages() && addresses.len() > 0 =>
-				{
-					let first_node_addresses = Some(addresses.to_vec());
+				Some((features, addresses)) if features.supports_onion_messages() => {
 					Ok(OnionMessagePath {
 						intermediate_nodes: vec![],
 						destination,
-						first_node_addresses,
+						first_node_addresses: addresses.to_vec(),
+					})
+				},
```

Comment:
Maybe we should not do this new always-generate-DC-event behavior if the node is announced, actually?